### PR TITLE
Fix [#3553] Add_http_if_no_scheme in domain in genspider

### PR DESCRIPTION
**Pull request by ThunderMind2019**
Fixed [#3553] 
before passing the domain, use built-in function from utils.url.add_http_if_no_scheme to add the http:// before the domain.
Updated template files by removing hard coded values.

### CODE REVIEW

File: scrapy/templates/spiders/basic.tmpl [hunk 1]
```diff
@@ -5,7 +5,7 @@ import scrapy
 class $classname(scrapy.Spider):
     name = '$name'
     allowed_domains = ['$domain']
-    start_urls = ['http://$domain/']
+    start_urls = ['$domain/']
```
[the remark this answers is not in the document]
Comment (ThunderMind2019) [reply scrapy/templates/spiders/basic.tmpl]: yeah got that, i that case we may also need to trim `http://` if user adds it.

[the remark this answers is not in the document]
Comment (maramsumanth) [reply scrapy/templates/spiders/basic.tmpl]: But why are you adding @ThunderMind2019 , according to the syntax we should only type domain name in place of ```<domain>``` in ```scrapy genspider <domain>```, not the whole URL.

[the remark this answers is not in the document]
Comment (ThunderMind2019) [reply scrapy/templates/spiders/basic.tmpl]: @maramsumanth There are Multiple reason, we shouldn't hard code values, second it the end user just randomly copy paste link, we should be clear on our end that our code stays safe as much as possible.
I think, things that we can control on our end, that dont change user's performance or any of his/her data, should be handled.

[the remark this answers is not in the document]
Comment (maramsumanth) [reply scrapy/templates/spiders/basic.tmpl]: If user wants a different schema other than ```http/https```, shouldn't our code be modified?

[the remark this answers is not in the document]
Comment (ThunderMind2019) [reply scrapy/templates/spiders/basic.tmpl]: That is a very rare case and in this case user should manually update the code as he/she knows the prefix (which is different than regular ones)

**Pull request closed without merging**